backend/utils/geo/geo.go

```diff
@@ -32,8 +32,8 @@ func GetIPLocation(ip, lang string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	if lang == "en" {
-		return geoLocation.Country.En + " " + geoLocation.Province.En, nil
+	if lang == "zh" {
+		return geoLocation.Country.Zh + " " + geoLocation.Province.Zh, nil
 	}
-	return geoLocation.Country.Zh + " " + geoLocation.Province.Zh, nil
+	return geoLocation.Country.En + " " + geoLocation.Province.En, nil
 }
```

Comment:
The code has two main issues:
1. In the second `if` statement, the return values are not used correctly.
2. The function does not handle invalid input gracefully.

### Changes suggested:
```go
func GetIPLocation(ip, lang string) (string, error) {
	if err != nil {
		return "", err
	}
	switch lang {
	case "en":
		return geoLocation.Country.En + " " + geoLocation.Province.En, nil
	case "zh":
		return geoLocation.Country.Zh + " " + geoLocation.Province.Zh, nil
	default:
		return "", fmt.Errorf("Unsupported language: %s", lang)
	}
}

// If 'geoLocation' is not defined elsewhere in this file, you should define it here to avoid compile errors.
type GeoLocation struct {
	Country struct {
		En string
		Zh string
	} Province struct {
		En string
		Zh string
	}
}
```

These changes address both issues by using a switch statement for better readability and proper handling of unsupported languages. Additionally, I've added missing type definitions and included a default case to indicate an unsupported language. This ensures that the function behaves as expected across different scenarios.